### DISCUSSION TITLE
I am having an issue when trying to import the CSV file.

### DISCUSSION THREAD

**Discussion by mclemens8** [BennyThadikaran/stock-pattern#86] [Q&A]
Hi, 

I am trying to important the CVS file but it just simply isnt working. I downloaded the data, formatted it to be correct, and I still cant get it to work. It keeps returning "file not found" and then the cvs. Let me know what you think, I will attach the CSV. 

Here is the error, (only a few lines) 

[00:00<?, ?it/s]File not found: C:\Users\mclemens1\stock-pattern\ï»¿date,close,open,high,low,volume.csv
  2%|█▌                                                                                 | 1/54 [00:02<02:16,  2.57s/it]File not found: C:\Users\mclemens1\stock-pattern\2022-02-23,20.21,21.88,22,20.19,36958180.csv
File not found: C:\Users\mclemens1\stock-pattern\2022-02-22,21.51,21.55,22.19,21.35,34908640.csv
[CCL.csv](https://github.com/user-attachments/files/17806399/CCL.csv)

Let me know what the issue is,

Thank you!

**Comment by BennyThadikaran** [comment]
Your watchlist is setup wrong. You have provided a OHLC file for the watchlist. So each line of OHLC data is treated as a symbol name.

Thats the reason for the weird file name below.

C:\Users\mclemens1\stock-pattern\2022-02-23,20.21,21.88,22,20.19,36958180.csv

Also CCL.csv must be lowercase.

You can try 

`py init.py --sym ccl.csv -p all`

If above is working well, create a watchlist.txt file with `ccl` on the first line and any other stocks you want to add (one on each line)

ccl
goog
gs

Run it like as below

`py init.py -f watchlist.txt -p all`

You can also add watchlist.txt in user.json.

I have not run the CCL.csv but it seems fine.

If you encounter any issues with parsing the date let me know.




On 18 November 2024 21:21:15 UTC, mclemens8 ***@***.***> wrote:
>Hi, 
>
>I am trying to important the CVS file but it just simply isnt working. I downloaded the data, formatted it to be correct, and I still cant get it to work. It keeps returning "file not found" and then the cvs. Let me know what you think, I will attach the CSV. 
>
>Here is the error, (only a few lines) 
>
>[00:00<?, ?it/s]File not found: C:\Users\mclemens1\stock-pattern\ï»¿date,close,open,high,low,volume.csv
>  2%|█▌                                                                                 | 1/54 [00:02<02:16,  2.57s/it]File not found: C:\Users\mclemens1\stock-pattern\2022-02-23,20.21,21.88,22,20.19,36958180.csv
>File not found: C:\Users\mclemens1\stock-pattern\2022-02-22,21.51,21.55,22.19,21.35,34908640.csv
>[CCL.csv](https://github.com/user-attachments/files/17806399/CCL.csv)
>
>
>Let me know what the issue is,
>
>
>Thank you!
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/BennyThadikaran/stock-pattern/discussions/86
>You are receiving this because you are subscribed to this thread.
>
>Message ID: ***@***.***>
Regards,
Benny Thadikaran

**Reply by mclemens8** [reply]
Hi! Funny you said I may find a parsing issue, I just got this issue 

C:\Users\mclemens1\stock-pattern>py init.py -f watchlist.txt -p all
Scanning `ALL` patterns on `daily`. Press Ctrl - C to exit
  0%|                                                                                            | 0/1 [00:00<?, ?it/s]ccl: Error loading file - ValueError("Missing column provided to 'parse_dates': '\ufeffDate'")
100%|████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.36it/s]
No patterns detected

**Comment by BennyThadikaran** [comment]
I was away from my PC when I first replied you. On second check there are a few issues with CCL.csv

1. You need to lower case the file names. (This is done because its tough to remember the exact case of filenames for 100+ files)
2. Your column names are all lower cased. You need it like `Date,Close,Open,High,Low,Volume`
3. If you use `setup-config.py`, make sure your data folder only has CSV files.

Finally, run `py init.py -p batu --sym ccl`. There is a nice BAT pattern detected on this one. :smile: 

Below is the edited OHLC file and the minimum config you will need to get going.

[ccl.csv](https://github.com/user-attachments/files/17810790/ccl.csv)

```json
{
  "DATA_PATH": "/home/benny/Desktop/ohlc"
}
```

**Reply by mclemens8** [reply]
Awesome, ive figured it out so far! My only issue now is that when i select "all" it doesnt detect any patterns, but when i select "batu" is does pick up the formation.

**Reply by BennyThadikaran** [reply]
I am working on a newer UI interface that will make the options more clear. Would you like to try it out? 

Download this to your project folder and rename it to `test.py` and run it. 

It will ask you to select a pattern and does nothing further. Just the user interface.

You can use `arrow keys` or `j` and `k` to navigate. You can also type the number to select the option. 

[test.txt](https://github.com/user-attachments/files/17827406/test.txt)

**Reply by mclemens8** [reply]
sure! ill test it for you, thank you

**Comment by BennyThadikaran** [comment]
The harmonic patterns and trendlines are currently not included in ALL, BULL or BEAR. It would produce too many results to be useful.

There is also some overlap since AB=CD is part of all harmonic patterns.

I will make it clear in the next update.

You can meanwhile go through the wiki section for more documentation.
Regards,
Benny Thadikaran

**Reply by mclemens8** [reply]
So, I should only choose specific types?

**Reply by BennyThadikaran** [reply]
ALL, Bull and Bear does not scan for Harmonic patterns and Trendlines. Those have to be scanned separately.

Bull and Bear does not scan for Triangles. I dont have a robust way to differentiate Ascending, Descending and Symmetric candles. (I leave it to the user to decide)

I am tightening the rules around Harmonic patterns, so by next update it should be added to the groups.

**Reply by mclemens8** [reply]
Ah I see, 

I was thinking, because a triangle needs to have coordinates for three points, why not measure each angle and make them variables so a1 = … a2 = … a3=… and then have an if statement like if a1 is < a2 then the triangle projected is an ascending triangle. And you can do the same for a descending triangle. Or have an else statement. Then you’d be able to incorporate them into bull and bear.

**Reply by BennyThadikaran** [reply]
I've been meaning to get some help in this area. Maybe you can provide some fresh perspective on this or anyone who might be reading this.

I use the slope of the line to check if Ascending or Descending. If the slope is 0, its a straight line, above 0 is ascending line, below 0 is descending line. 

Since markets seldom form straight lines, i allow some threshold like -0.5 to 0.5 to determine a straight line. This works well for lines on the same chart. But when comparing slopes from different charts, a 0.5 can look much steeper on some charts and almost straight on others. 

This has to do with the price scale and the date range involved for the various charts.

I have an idea but not sure if it will work. I apply a scaling factor to the price with a base value of 1000. The lines would be of different lengths so those need to be scaled as well. 

Once this is done, calculate the slope and set some threshold values. It a bit of work, so i left this for the time being to test when i have plenty of time at hand :smiley: 

Run the code below, you will see what i mean.

```python
import matplotlib.pyplot as plt
import pandas as pd

df1 = pd.DataFrame(
    [100, 102, 101, 105, 110],
    index=pd.date_range(start="2024-01-01", periods=5),
    columns=pd.Index(["Close"]),
)

df2 = pd.DataFrame(
    [20000, 20200, 19800, 20500, 21000],
    index=pd.date_range(start="2024-01-01", periods=5),
    columns=pd.Index(["Close"]),
)

target_value = 1000

# Scale the Close price of each stock to 1000
df1["d1"] = df1.Close * (target_value / df1.Close.iloc[0])
df2["d2"] = df2.Close * (target_value / df2.Close.iloc[0])

# Plot the scaled data
plt.plot(df1.index, df1.d1, label="Stock 1 (Scaled)", marker="o")

plt.plot(df2.index, df2.d2, label="Stock 2 (Scaled)", marker="o")

plt.title("Stocks Scaled to 1000")
plt.xlabel("Date")
plt.ylabel("Scaled Price")
plt.legend()
plt.show()
```

**Comment by BennyThadikaran** [comment]
@mclemens8 I just released version 4.0.4. It addresses some issues you raised. 

For example, 

`all` now includes everything including Triangles, trendlines, harmonic patterns. 

`bull` and `bear` now includes harmonic patterns.
